GitCommands/Git/Extensions/GitRevisionExtensions.cs

```diff
@@ -0,0 +1,29 @@
+﻿using System;
+
+namespace GitCommands.Git.Extensions
+{
+    public static class GitRevisionExtensions
+    {
+        public static bool IsArtificial(this string sha1)
+        {
+            return sha1 == GitRevision.UnstagedGuid ||
+                   sha1 == GitRevision.IndexGuid;
+        }
+
+        public static string ToShortSha(this string sha1)
+        {
+            if (string.IsNullOrWhiteSpace(sha1))
```

Comment:
I rather keep the current behavior and not throw exception at empty
This is a situation where Assert are preferred

Reply:
an alternative would be transforming `null` into `""`

Reply:
It should be OK if null is handled as "", sometimes the intention. (I am struggling some with related in FileStatusList now)